### PR TITLE
LPS-122337 create unit tests

### PR DESCRIPTION
**Pull request by marcosapmf**
Related issue: https://issues.liferay.com/browse/LPS-122337

### CODE REVIEW

**Comment by marcosapmf** [comment]
ci:test:sf

**Comment by marcosapmf** [comment]
ci:help

**Comment by marcosapmf** [comment]
ci:test:sf

**Comment by marcosapmf** [comment]
ci:test:relevant

**Comment by marcosapmf** [comment]
ci:test:forms

**Comment by marcosapmf** [comment]
ci:test:data-engine

**Comment by marcosapmf** [comment]
ci:test:app-builder

**Comment by marcosapmf** [comment]
ci:test:sf

**Comment by marcosapmf** [comment]
ci:test:relevant

**Comment by marcosapmf** [comment]
ci:test:sf

**Comment by marcosapmf** [comment]
ci:test:relevant

**Comment by marcelabc** [comment]
Just started reviewing :) <br><br>:octocat: *Sent from [GH](http://nodegh.io).*

File: modules/apps/data-engine/data-engine-taglib/src/main/resources/META-INF/resources/data_layout_builder/js/data-layout-builder/DataLayoutBuilder.es.js [hunk 1]
```diff
@@ -500,9 +500,56 @@ class DataLayoutBuilder extends React.Component {
...
+				...settingsContext,
+				pages: settingsContextPagesVisitor.mapFields(
+					(settingsField) => {
+						if (settingsField.type === 'options') {
+							const {value} = settingsField;
+							const newValue = {};
+
+							Object.keys(value).forEach((locale) => {
+								newValue[locale] = value[locale].filter(
+									({value}) => value !== ''
```
Comment (marcosapmf) [inline modules/apps/data-engine/data-engine-taglib/src/main/resources/META-INF/resources/data_layout_builder/js/data-layout-builder/DataLayoutBuilder.es.js:527]: Alterar para (localizedValue) => localizedValue.value !== ''

**Comment by marcosapmf** [comment]
Moved to https://github.com/liferay-data-engine/liferay-portal/pull/34

**Pull request closed without merging**